src/libraries/Microsoft.Extensions.Configuration.FileExtensions/tests/FileConfigurationProviderTest.cs

```diff
@@ -34,6 +32,40 @@ public void ProviderDisposesChangeTokenRegistration()
             Assert.Empty(changeToken.Callbacks);
         }
 
+        [Theory]
+        [InlineData(@"C:\path\to\configuration.txt")]
```

Comment:
Nit, could we use random GUIDs for the path? I feel like this directory could exist on a machine and not fail with `DirectoryNotFoundException` but instead throw `FileNotFoundException` causing this test to fail?

Reply:
Updated this to use `Guid` instead, and added proper tests for:

- `InvalidDataException`
- `FileNotFoundException`
- `DirectoryNotFoundException`